odoo/addons/base/ir/ir_cron.py

```diff
@@ -171,41 +171,53 @@ def _process_job(cls, job_cr, job, cron_cr):
 
     @classmethod
     def _acquire_job(cls, db_name):
-        # TODO remove 'check' argument from addons/base_action_rule/base_action_rule.py
-        """ Try to process one cron job.
+        """ Try to process all cron jobs.
 
         This selects in database all the jobs that should be processed. It then
         tries to lock each of them and, if it succeeds, run the cron job (if it
         doesn't succeed, it means the job was already locked to be taken care
         of by another thread) and return.
 
-        If a job was processed, returns True, otherwise returns False.
+        Returns ``None`` in case of success; otherwise returns a string value
+        describing the failure.
         """
         db = odoo.sql_db.db_connect(db_name)
         threading.current_thread().dbname = db_name
+        error = None
         jobs = []
         try:
             with db.cursor() as cr:
-                # Make sure the database we poll has the same version as the code of base
-                cr.execute("SELECT 1 FROM ir_module_module WHERE name=%s AND latest_version=%s", ('base', BASE_VERSION))
-                if cr.fetchone():
+                # Make sure the database has the same version as the code of
+                # base and that no module must be installed/upgraded/removed
+                cr.execute("SELECT 1 FROM ir_module_module WHERE name=%s AND latest_version=%s",
+                           ['base', BASE_VERSION])
+                version = cr.fetchone()
+                cr.execute("SELECT 1 AS c FROM ir_module_module WHERE state LIKE %s", ['to %'])
+                changes = cr.fetchone()
+                if not version:
+                    _logger.warning('Skipping database %s as its base version is not %s.', db_name, BASE_VERSION)
+                    error = 'version'
+                elif changes:
+                    _logger.warning('Skipping database %s because of modules to install/upgrade/remove.', db_name)
+                    error = 'state'
+                else:
                     # Careful to compare timestamps with 'UTC' - everything is UTC as of v6.1.
                     cr.execute("""SELECT * FROM ir_cron
                                   WHERE numbercall != 0
                                       AND active AND nextcall <= (now() at time zone 'UTC')
                                   ORDER BY priority""")
                     jobs = cr.dictfetchall()
-                else:
-                    _logger.warning('Skipping database %s as its base version is not %s.', db_name, BASE_VERSION)
         except psycopg2.ProgrammingError, e:
             if e.pgcode == '42P01':
                 # Class 42 — Syntax Error or Access Rule Violation; 42P01: undefined_table
                 # The table ir_cron does not exist; this is probably not an OpenERP database.
                 _logger.warning('Tried to poll an undefined table on database %s.', db_name)
+                error = 'undefined table'
```

Comment:
Returning single words seems simpler in terms of API, for example `'database'` to indicate that this database is in an unknown state.